src/kvstore/kvstore_dist.h

```diff
@@ -185,7 +195,7 @@ class KVStoreDist : public KVStoreLocal {
           RunContext rctx, Engine::CallbackOnComplete cb) {
         // convert to ps keys
         size_t size = recv_buf.shape().Size();
-        PSKV& pskv = EncodeKey(key, size);
+        PSKV& pskv = EncodeKey(key, size, false);
```

Comment:
Was the `false` flag always not there? Was there a bug?

Reply:
No, I added a boolean to this function representing push or pull, since they now use different kvs.